test/extended/openstack/loadbalancers.go

```diff
@@ -0,0 +1,239 @@
+
+	oc := exutil.NewCLI("openstack")
+	var loadBalancerClient *gophercloud.ServiceClient
+	var networkClient *gophercloud.ServiceClient
+	var clientSet *kubernetes.Clientset
+
+	g.BeforeEach(func() {
+
+		network, err := oc.AdminConfigClient().ConfigV1().Networks().Get(context.Background(), "cluster", metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if network.Status.NetworkType == "Kuryr" {
+			e2eskipper.Skipf("Test not applicable for Kuryr NetworkType")
+		}
+
+		g.By("preparing openstack client")
+		loadBalancerClient, err = client(serviceLoadBalancer)
+		o.Expect(err).NotTo(o.HaveOccurred(), "Error creating an openstack LoadBalancer client")
+		networkClient, err = client(serviceNetwork)
+		o.Expect(err).NotTo(o.HaveOccurred(), "Error creating an openstack network client")
+
+		g.By("preparing openshift dynamic client")
+		clientSet, err = e2e.LoadClientset()
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	// https://issues.redhat.com/browse/OSASINFRA-2753
+	g.It("should create an UDP Amphora LoadBalancer when an UDP svc with type:LoadBalancer is created on Openshift", func() {
+
+		g.By("Checking loadBalancer provider")
+		cloudProviderConfig, err := getConfig(oc.AdminKubeClient(),
+			"openshift-config",
+			"cloud-provider-config",
+			"config")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		lbProvider, err := getClusterLoadBalancerSetting("lb-provider", cloudProviderConfig)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if lbProvider != "amphora" {
+			e2eskipper.Skipf("Test not applicable for LoadBalancer provider different to Amphora: %q", lbProvider)
+		}
+
+		g.By("Creating Openshift resources")
+		// Reference on test/extended/testdata/openstack/udplb-default-manifests.yaml
+		labels := map[string]string{"app": "udp-lb-default-dep"}
+		depName := "udp-lb-default-dep"
+		svcName := "udp-lb-default-svc"
+		testDeployment := e2edeployment.NewDeployment(depName, 2, labels, "udp-test",
+			imageutils.GetE2EImage(imageutils.Agnhost), appsv1.RollingUpdateDeploymentStrategyType)
+		testDeployment.Spec.Template.Spec.SecurityContext = e2epod.GetRestrictedPodSecurityContext()
+		testDeployment.Spec.Template.Spec.Containers[0].SecurityContext = e2epod.GetRestrictedContainerSecurityContext()
+		testDeployment.Spec.Template.Spec.Containers[0].Args = []string{"netexec", "--udp-port=8081"}
+		testDeployment.Spec.Template.Spec.Containers[0].Ports = []v1.ContainerPort{{
+			ContainerPort: 8081,
+			Protocol:      "UDP",
+		}}
+		deployment, err := clientSet.AppsV1().Deployments(oc.Namespace()).Create(context.TODO(),
+			testDeployment, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = e2edeployment.WaitForDeploymentComplete(clientSet, deployment)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		jig := e2eservice.NewTestJig(clientSet, oc.Namespace(), svcName)
+		jig.Labels = labels
+		svc, err := jig.CreateLoadBalancerService(120*time.Second, func(svc *v1.Service) {
```

Comment:
That timeout should probably go to some constant. And I'm not sure it's not too low for some environments. Default timeout for pods in e2e tests is 5 minutes.

Reply:
Done, thanks!